packages/core/src/modules/indy/repository/AnonCredsCredentialDefinitionRecord.ts

```diff
@@ -1,39 +1,47 @@
+import type { DidIndyNamespace } from '../../../utils/indyIdentifiers'
 import type { CredDef } from 'indy-sdk'
 
 import { BaseRecord } from '../../../storage/BaseRecord'
-import { didFromCredentialDefinitionId } from '../../../utils/did'
+import { getQualifiedIdentifier } from '../../../utils/indyIdentifiers'
+import { CredentialDefinitionTransformer } from '../../../utils/transformers'
+import { uuid } from '../../../utils/uuid'
 
 export interface AnonCredsCredentialDefinitionRecordProps {
   credentialDefinition: CredDef
+  didIndyNamespace: DidIndyNamespace
+  schemaSeqNo: number
 }
 
-export type DefaultAnonCredsCredentialDefinitionTags = {
-  credentialDefinitionId: string
-  issuerDid: string
-  schemaId: string
-  tag: string
-}
-
-export class AnonCredsCredentialDefinitionRecord extends BaseRecord<DefaultAnonCredsCredentialDefinitionTags> {
+export class AnonCredsCredentialDefinitionRecord extends BaseRecord {
   public static readonly type = 'AnonCredsCredentialDefinitionRecord'
   public readonly type = AnonCredsCredentialDefinitionRecord.type
-  public readonly credentialDefinition!: CredDef
+
+  @CredentialDefinitionTransformer()
+  public readonly credentialDefinition!: CredDef & { didIndyNamespace?: DidIndyNamespace; schemaSeqNo?: number }
```

Comment:
I don't think we need this. We can extract the did indy namespace from the identifier and the schemaSeqNo from the schemaID property

```suggestion
  public readonly credentialDefinition!: CredDef
```

Reply:
will also double-check whether/how to refactor this out.

Reply:
Might be missing sth here but how do I get the schemaSeqNo from the schemaId?

Reply:
as discussed (just for the record) cannot get schemaSeqNo from schemaId but from extra ledger request to resolve the schema via the credDef

Reply:
Left schemaSeqNo in for now as I'm unsure where to best do the ledger request here to resolve it from the cred def.